appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java

```diff
@@ -474,11 +475,14 @@ public WeldApplicationContainer load(WeldContainer container, DeploymentContext
         String archiveName = !isSubArchive? appInfo.getName() : archive.getName();
         int suffixIdx = archiveName.lastIndexOf('-');
         if(isSubArchive && suffixIdx > 0) {
-            archiveName = archiveName.substring(0, suffixIdx);
-            if(!context.getArchiveHandler().getArchiveType().isEmpty()) {
-                archiveName = String.format("%s.%s", archiveName, context.getArchiveHandler().getArchiveType());
+            String versionStr = archiveName.substring(suffixIdx + 1, archiveName.length());
+            if(versionStr.matches("^[0-9]+\\..*")) {
```

Comment:
This checks for digit followed by dot followed by anything else. Windows shouldn't be an issue because the backslash is used as an escape sequence for the dot only.